src/constraints/outage_constraints.jl

```diff
@@ -143,14 +154,73 @@ function add_MG_fuel_burn_constraints(m,p)
     )
     
     @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps],
-        m[:dvMGMaxFuelCost][s] >= sum( MGFuelCost[t, s, tz] for t in p.techs.gen )
+        m[:dvMGGenMaxFuelCost][s] >= sum( MGFuelCost[t, s, tz] for t in p.techs.gen )
     )
     
-    @expression(m, ExpectedMGFuelCost,
-        sum( m[:dvMGMaxFuelCost][s] * p.s.electric_utility.outage_probabilities[s] for s in p.s.electric_utility.scenarios )
+    @expression(m, ExpectedMGGenFuelCost,
+        sum( m[:dvMGGenMaxFuelCost][s] * p.s.electric_utility.outage_probabilities[s] for s in p.s.electric_utility.scenarios )
     )
+
+    m[:ExpectedMGFuelUsed] += ExpectedMGGenFuelUsed
+    m[:ExpectedMGFuelCost] += ExpectedMGGenFuelCost
 end
 
+function add_MG_CHP_fuel_burn_constraints(m, p; _n="")
+    # Fuel burn slope and intercept
+    fuel_burn_full_load = 1.0 / p.s.chp.electric_efficiency_full_load  # [kWt/kWe]
+    fuel_burn_half_load = 0.5 / p.s.chp.electric_efficiency_half_load  # [kWt/kWe]
+    fuel_burn_slope = (fuel_burn_full_load - fuel_burn_half_load) / (1.0 - 0.5)  # [kWt/kWe]
+    fuel_burn_intercept = fuel_burn_full_load - fuel_burn_slope * 1.0  # [kWt/kWe_rated]
+  
+    # Conditionally add dvFuelBurnYIntercept if coefficient p.FuelBurnYIntRate is greater than ~zero
+    if fuel_burn_intercept > 1.0E-7
+        #Constraint (1c1): Total Fuel burn for CHP **with** y-intercept fuel burn and supplementary firing
+        @constraint(m, MGCHPFuelBurnCon[t in p.techs.chp, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps],
+            m[Symbol("dvMGFuelUsed"*_n)][t,s,tz]  == p.hours_per_time_step * 
+                m[Symbol("dvMGCHPFuelBurnYIntercept"*_n)][s,tz] +
+                sum(p.production_factor[t,tz+ts-1] * fuel_burn_slope * m[Symbol("dvMGRatedProduction"*_n)][t,s,tz,ts]
+                    for ts in 1:p.s.electric_utility.outage_durations[s])
+        )
+
+        #Constraint (1d): Y-intercept fuel burn for CHP across the scenario outage time steps
```

Comment:
I'm noticing that this CHP fuel burn intercept usage is different than for generator (in outages and not). The CHP fuel_burn_intercept is multiplied by dvSize to get dvMGCHPFuelBurnYIntercept, which seems correct to me based on how fuel_burn_intercept is calculated. But it's making me think that this is handled incorrectly for generator, probably a result of changing the gen fuel inputs to efficiencies. Gen slope and intercept are calculated the same way as for CHP (plus unit conversion to gal) so that intercept value needs to be multiplied by size, however it is used the way that the old input fuel_intercept_gal_per_hr (which didn't need to be multiplied by size) was used. Kinda confusing, am I making sense?

Reply:
Or maybe it's just late and I can't think straight because wouldn't the generator fuel input change have broken tests if this was the case?

Reply:
Hmm, yeah I think you're right. I'll take a look.